### ISSUE TITLE
"file could not be opened successfully" on a v2.LR.historical_0101 tar ball

### ISSUE THREAD

**Issue by chengzhuzhang** [E3SM-Project/zstash#217]
I'm trying to extract the v2.LR.historical_0101 run from NERSC HPSS archive and ran into "file could not be opened successfully" error on zstash/00005.tar when beginning extract `archive/atm/hist/v2.LR.historical_0101.eam.h0.1978-05.nc`. 

To reproduce:

```
source /global/common/software/e3sm/anaconda_envs/load_latest_e3sm_unified_cori-haswell.sh
zstash extract  --hpss=/home/f/forsyth/E3SMv2/v2.LR.historical_0101 "archive/atm/hist/*h0.197[8-9]*"
```
Not sure how to trouble shooting, maybe it's a problem in the archive?

The Traceback is as follows:

```
INFO: Opening tar archive zstash/000005.tar
Traceback (most recent call last):
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/bin/zstash", line 10, in <module>
    sys.exit(main())
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/lib/python3.9/site-packages/zstash/main.py", line 64, in main
    extract()
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/lib/python3.9/site-packages/zstash/extract.py", line 45, in extract
    failures: List[FilesRow] = extract_database(args, cache, keep_files)
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/lib/python3.9/site-packages/zstash/extract.py", line 275, in extract_database
    failures = extractFiles(matches, keep_files, keep, cache)
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/lib/python3.9/site-packages/zstash/extract.py", line 444, in extractFiles
    tar: tarfile.TarFile = tarfile.open(tfname, "r")
  File "/global/common/software/e3sm/anaconda_envs/base/envs/e3sm_unified_1.7.0_nompi/lib/python3.9/tarfile.py", line 1625, in open
    raise ReadError("file could not be opened successfully")

```

**Comment by chengzhuzhang** [comment]
After consulting @golaz, a failed extraction leave an incomplete tar file (i.e. 00005.tar in this case) on the local disk under `zstash/`. and zstash will not retrieve it from tape the second time because it thinks the file is already there. To remove the tar file on disk (under zstash/ sub-directory) and try again resolved the problem. 

An earlier [feature request](https://github.com/E3SM-Project/zstash/issues/198) on supporting retry would be helpful to handle extraction failure due to some HPSS system problems.

**Comment by forsyth2** [comment]
Resolved by #218.
